lib/discourse_livestream/topic_extension.rb

```diff
@@ -8,6 +8,13 @@ module TopicExtension
       has_one :topic_chat_channel,
               class_name: "DiscourseLivestream::TopicChatChannel",
               dependent: :destroy
+
+      after_update do
+        if topic_chat_channel.present? &&
+             topic_chat_channel&.chat_channel&.chatable_id != category_id
```

Comment:
is the separate `presence` check needed?

```suggestion
        if topic_chat_channel&.chat_channel&.chatable_id != category_id
```

Reply:
I'll revert